app/services/ReviewService.scala

```diff
@@ -64,31 +64,61 @@ class ReviewService @Inject() (publishedService: PublishedService, repository: A
   def changeStatus(id: String, currentStatus: String, info: ApprovalProcessStatusChange): Future[RequestOutcome[Unit]] = {
 
     def getContentToUpdate: Future[Option[ApprovalProcess]] = repository.getById(id) map {
-        case Right(process) => if (process.meta.status == currentStatus) Some(process) else None
-        case _ => None
-      }
+      case Right(process) =>
+        if (process.meta.status == currentStatus) {
+          Some (process)
+        }
+        else {
+          logger.warn(s"Invalid Process Status Change requested for process $id: " +
+            s"Expected Status: '$currentStatus' Status Found: '${process.meta.status}' Desired Status: ${info.status} ")
+          None
+        }
+      case _ =>
+        logger.warn(s"ChangeStatus - process id ($id) not found.")
+        None
+    }
 
     def publishIfRequired(approvalProcess: ApprovalProcess): Future[RequestOutcome[Unit]] = info.status match {
-        case StatusApprovedForPublishing =>
-          publishedService.save(id, approvalProcess.process) map {
-            case Right(_) => Right(())
-            case Left(errors) => Left(errors)
-          }
-        case _ => Future.successful(Right(()))
-      }
+      case StatusApprovedForPublishing =>
+        publishedService.save(id, approvalProcess.process) map {
+          case Right(_) => Right(())
+          case Left(errors) =>
+            logger.error(s"Failed to publish $id - $errors")
+            Left(errors)
+        }
+      case _ => Future.successful(Right(()))
+    }
 
     getContentToUpdate flatMap {
```

Comment:
You've converted to Option return and lost potential errors which may have occured in the RequestOutcome calls, sticking to RequestOutcome throughout will ensure fast fail and errors returned

Reply:
updated